web_widget_html_markdown/readme/DESCRIPTION.rst

```diff
@@ -0,0 +1,7 @@
+This module adds a new widget for Html field in form view on Odoo:
+
+- In readonly mode, it renders normal Html
+- In write mode, use markdown or normal Html editor, with switcher to 
+  allow to convert/switch bidirectionally between Html and Markdown. Data will always 
+  be saved in Html.
```

Comment:
Not so clear, I would say:

```
- In readonly mode, the widget renders as Html
- In editable mode, there is a switcher to choose if the source should be HTML or Markdown. If HTML, it shows the HTML edit widget. If Markdown, it shows a markdown widget. When done editing, it will save in the field as Html, no matter which widget is chosen.
- In markdown mode, the markdown is saved embedded within the HTML so that it can be shown again on subsequent edits of the field and there is no loss from backconversion of Html to Markdown.
- When switching the widget from Markdown to HTML, the Markdown source is lost.
- When switching the widget from HTML back to Markdown, the Showdown.JS backconversion is used to generate a reasonable representation of the HTML in markdown. This is where some information loss can occur, so the user should not needlessly switch from HTML to Markdown and back - in general, choose one way to edit your field content, and stick to it.

Reply:
And add something like:

_Why this module when `web_widget_text_markdown` already exists? Mainly because it is designed to work on `Text` fields, and most fields within Odoo are HTML fields for example: mail content, project.task description etc. Although you could develop a custom module that overrides these fields to be Text fields and render them on-screen as HTML, this rendering would not be done when the field is for example used in sending out a mail or fetching the field content through Odoo API. In these cases this widget comes in handy because it is plug-and-play: the field type and content remains as HTML, but you can choose to edit it with a markdown widget._

Reply:
Adding bulletpoint verbatim after checking.